### PR TITLE
Inline documentation

### PR DESCRIPTION
**Pull request by Gusinuhe**
### Description

Inline documentation for all public, internal, and protected members.

**Fixes**: Fix missing API documentation

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

**Comment by SimonTheSourcerer** [comment]
Remember, conventional commit messages, but we definitly have to improve error message output :D

**Comment by aleksei-korolev** [comment]
Do we really need to comment internal stuff, the one which is fairly trivial?

**Comment by Gusinuhe** [comment]
> Remember, conventional commit messages, but we definitly have to improve error message output :D

Sure but, should we follow the conventional commit guidelines even for commits that will be squashed?

File: Editor/TestTools/EditorImguiTester/EditorImguiTestFinishedEventArgs.cs [hunk 1]
```diff
@@ -3,10 +3,13 @@
 namespace Innoactive.CreatorEditor.TestTools
 {
     /// <summary>
-    /// Event args for event which is fired when Editor IMGUI test finishes it's execution.
+    /// Event args for event which is fired when a <see cref="IEditorImguiTest"/> test finishes it's execution.
```
Comment (Maxsiom) [inline Editor/TestTools/EditorImguiTester/EditorImguiTestFinishedEventArgs.cs:6]:
```suggestion
    /// Event args for event which is fired when a <see cref="IEditorImguiTest"/> test finishes its execution.
```

File: Editor/TestTools/EditorImguiTester/EditorImguiTestFinishedEventArgs.cs [hunk 2]
```diff
@@ -3,10 +3,13 @@
...
     /// </summary>
     internal class EditorImguiTestFinishedEventArgs : EventArgs
     {
+        /// <summary>
+        /// Result from last <see cref="IEditorImguiTest"/>.
```
Comment (Maxsiom) [inline Editor/TestTools/EditorImguiTester/EditorImguiTestFinishedEventArgs.cs:11]:
```suggestion
        /// Result from the last <see cref="IEditorImguiTest"/>.
```

File: Runtime/ActivationStateChangedEventArgs.cs [hunk 3]
```diff
@@ -2,8 +2,14 @@
 
 namespace Innoactive.Creator.Core
 {
+    /// <summary>
+    /// Event that fired when the current stage changes.
```
Comment (Maxsiom) [inline Runtime/ActivationStateChangedEventArgs.cs:6]:
```suggestion
    /// Event that is fired when the current stage changes.
```

File: Runtime/Attributes/DisplayNameAttribute.cs [hunk 4]
```diff
@@ -8,6 +8,9 @@ namespace Innoactive.Creator.Core.Attributes
     [AttributeUsage(AttributeTargets.Property | AttributeTargets.Field | AttributeTargets.Class)]
     public class DisplayNameAttribute : Attribute
     {
+        /// <summary>
+        /// Name of training entity's property or field.
```
Comment (Maxsiom) [inline Runtime/Attributes/DisplayNameAttribute.cs:12]:
```suggestion
        /// Name of the training entity's property or field.
```

File: Runtime/CourseRunner.cs [hunk 5]
```diff
@@ -13,6 +13,9 @@ public static class CourseRunner
     {
         private class CourseRunnerInstance : MonoBehaviour
         {
+            /// <summary>
+            /// Reference to currently used <see cref="ICourse"/>.
```
Comment (Maxsiom) [inline Runtime/CourseRunner.cs:17]:
```suggestion
            /// Reference to the currently used <see cref="ICourse"/>.
```

File: Runtime/IBehaviorCollectionData.cs [hunk 6]
```diff
@@ -5,12 +5,12 @@
 namespace Innoactive.Creator.Core
 {
     /// <summary>
-    /// A data that contains list of behaviors.
+    /// A data that contains list of <see cref="IBehavior"/>s.
```
Comment (Maxsiom) [inline Runtime/IBehaviorCollectionData.cs:8]:
```suggestion
    /// A data that contains a list of <see cref="IBehavior"/>s.
```

File: Runtime/ICompletable.cs [hunk 7]
```diff
@@ -1,8 +1,18 @@
...
+    /// </summary>
     public interface ICompletable
     {
+        /// <summary>
+        /// True if this instance is already completed.
+        /// </summary>
         bool IsCompleted { get; }
+
+        /// <summary>
+        /// Forces this instance completion.
```
Comment (Maxsiom) [inline Runtime/ICompletable.cs:14]:
```suggestion
        /// Forces this instance's completion.
```

File: Runtime/IDescribedData.cs [hunk 8]
```diff
@@ -1,7 +1,13 @@
 namespace Innoactive.Creator.Core
 {
+    /// <summary>
+    /// Data structure with an <see cref="IStep"/>'s description.
+    /// </summary>
     public interface IDescribedData : IData
     {
+        /// <summary>
+        /// <see cref="IStep"/> description.
```
Comment (Maxsiom) [inline Runtime/IDescribedData.cs:9]:
```suggestion
        /// <see cref="IStep"/>'s description.
```

File: Runtime/INamedData.cs [hunk 9]
```diff
@@ -3,8 +3,14 @@
 
 namespace Innoactive.Creator.Core
 {
+    /// <summary>
+    /// Data structure with an <see cref="IStep"/>'s name.
+    /// </summary>
     public interface INamedData : IData
     {
+        /// <summary>
+        /// <see cref="IStep"/> name.
```
Comment (Maxsiom) [inline Runtime/INamedData.cs:12]:
```suggestion
        /// <see cref="IStep"/>'s name.
```

File: Editor/UI/Drawers/MetadataWrapperDrawer.cs [hunk 10]
```diff
@@ -228,10 +228,8 @@ private Rect DrawExtendableList(Rect rect, MetadataWrapper wrapper, Action<objec
                 return rect;
             }
 
-            Type elementType = (wrapper.Metadata[extendableListName] as ExtendableListAttribute.SerializedTypeWrapper).Type;
-
+            Type elementType = (wrapper.Metadata[extendableListName] as ExtendableListAttribute.SerializedTypeWrapper)?.Type;
```
Comment (SimonTheSourcerer) [inline Editor/UI/Drawers/MetadataWrapperDrawer.cs:231]: This just moves the NPE into DrawerLocator.GetInstantiatorDrawer which is actually it makes worse to find :P

File: Runtime/ActivationStateChangedEventArgs.cs [hunk 11]
```diff
@@ -2,8 +2,14 @@
 
 namespace Innoactive.Creator.Core
 {
+    /// <summary>
+    /// Event that is fired when the current stage changes.
+    /// </summary>
     public class ActivationStateChangedEventArgs : EventArgs
     {
+        /// <summary>
+        /// New stage.
```
Comment (SimonTheSourcerer) [inline Runtime/ActivationStateChangedEventArgs.cs:11]: stage or state?

File: Runtime/ActivationStateChangedEventArgs.cs [hunk 12]
```diff
@@ -2,8 +2,14 @@
 
 namespace Innoactive.Creator.Core
 {
+    /// <summary>
+    /// Event that is fired when the current stage changes.
```
Comment (SimonTheSourcerer) [inline Runtime/ActivationStateChangedEventArgs.cs:6]: current state or stage?

File: Runtime/ChildActivatedEventArgs.cs [hunk 13]
```diff
@@ -2,6 +2,7 @@
 
 namespace Innoactive.Creator.Core
 {
+    [Obsolete("This event is not used anymore.")]
```
Comment (SimonTheSourcerer) [inline Runtime/ChildActivatedEventArgs.cs:5]: Is there are alternative to use?

File: Runtime/ChildDeactivatedEventArgs.cs [hunk 14]
```diff
@@ -2,6 +2,7 @@
 
 namespace Innoactive.Creator.Core
 {
+    [Obsolete("This event is not used anymore.")]
```
Comment (SimonTheSourcerer) [inline Runtime/ChildDeactivatedEventArgs.cs:5]: Is there are alternative to use?

File: Runtime/CurrentChildChangedEventArgs.cs [hunk 15]
```diff
@@ -2,6 +2,7 @@
 
 namespace Innoactive.Creator.Core
 {
+    [Obsolete("This event is not used anymore.")]
```
Comment (SimonTheSourcerer) [inline Runtime/CurrentChildChangedEventArgs.cs:5]: Is there are alternative to use?

File: Runtime/CourseRunner.cs [hunk 16]
```diff
@@ -68,24 +74,12 @@ public void Execute()
...
         /// Returns true if the current <see cref="ICourse"/> is running.
         /// </summary>
-        public static bool IsRunning
-        {
-            get
-            {
-                return Current != null && Current.LifeCycle.Stage != Stage.Inactive;
-            }
-        }
+        public static bool IsRunning => Current != null && Current.LifeCycle.Stage != Stage.Inactive;
```
Comment (SimonTheSourcerer) [inline Runtime/CourseRunner.cs:82]: What is your opinion on these kind of getter? Checked our code style but they are not mentioned.

File: Runtime/CurrentChildChangedEventArgs.cs (hunk 15 above)
Comment (Gusinuhe) [reply Runtime/CurrentChildChangedEventArgs.cs:5]: Not really, it has not been used in the whole v1.x.
I am not even sure when or why was it used.

File: Runtime/ChildDeactivatedEventArgs.cs (hunk 14 above)
Comment (Gusinuhe) [reply Runtime/ChildDeactivatedEventArgs.cs:5]: Not really, it has not been used in the whole v1.x.
I am not even sure when or why was it used.

File: Runtime/ChildActivatedEventArgs.cs (hunk 13 above)
Comment (Gusinuhe) [reply Runtime/ChildActivatedEventArgs.cs:5]: Not really, it has not been used in the whole v1.x.
I am not even sure when or why was it used.

File: Runtime/ActivationStateChangedEventArgs.cs (hunk 12 above)
Comment (Gusinuhe) [reply Runtime/ActivationStateChangedEventArgs.cs:6]: Stage

File: Runtime/ActivationStateChangedEventArgs.cs (hunk 11 above)
Comment (Gusinuhe) [reply Runtime/ActivationStateChangedEventArgs.cs:11]: Stage

**Pull request merged**